beginner_source/quickstart/data_quickstart_tutorial.py

```diff
@@ -0,0 +1,186 @@
+#
+
+###############################################################
+# .. figure:: /_static/img/quickstart/typesdata.png
+#    :alt: typesdata
+#
+# Different data types require different python libraries to load and process such as `openCV <https://opencv.org/>`_ and `PIL <https://pillow.readthedocs.io/en/stable/reference/Image.html>`_ for images, `NLTK <https://www.nltk.org/>`_ and `spaCy <https://spacy.io/>`_ for text and `Librosa <https://librosa.org/doc/latest/index.html>`_ for audio.
+#
+# If not properly organized, code for processing data samples can quickly get messy and become hard to maintain. Since different model architectures can be applied to many data types, we ideally want our dataset code to be decoupled from our model training code. To this end, PyTorch provides a simple Datasets interface for linking managing collections of data.
+#
+# A whole set of example datasets such as Fashion MNIST that implement this interface are built into PyTorch extension libraries. These are useful for benchmarking and testing your models before training on your own custom datasets.
+#
+#  You can find some of them below.
+#
+#  - `Image Datasets <https://pytorch.org/docs/stable/torchvision/datasets.html>`_
+#  - `Text Datasets  <https://pytorch.org/text/datasets.html>`_
+#  - `Audio Datasets <https://pytorch.org/audio/datasets.html>`_
+#
+
+#################################################################
+# Iterating through a Dataset
+# -----------------
+#
+# Once we have a Dataset we can index it manually like a list `clothing[index]`.
+#
+# Here is an example of how to load the fashion MNIST dataset from torch vision.
+#
+
+from torch.utils.data import DataLoader
+from torchvision.io import read_image
+from torchvision import transforms, utils
+import pandas as pd
+import torch
+import os
+import torch
+from torch.utils.data import Dataset
+import torchvision.datasets as datasets
+import matplotlib.pyplot as plt
+import numpy as np
+
+clothing = datasets.FashionMNIST('data', train=True, download=True)
+labels_map = {0: 'T-Shirt', 1: 'Trouser', 2: 'Pullover', 3: 'Dress',
+              4: 'Coat', 5: 'Sandal', 6: 'Shirt', 7: 'Sneaker', 8: 'Bag', 9: 'Ankle Boot'}
+figure = plt.figure(figsize=(8, 8))
+cols, rows = 3, 3
+for i in range(1, cols*rows + 1):
+    sample_idx = np.random.randint(len(clothing))
+    img = clothing[sample_idx][0][0, :, :]
+    figure.add_subplot(rows, cols, i)
+    plt.title(labels_map[clothing[sample_idx][1]])
+    plt.axis('off')
+    plt.imshow(img, cmap='gray')
+plt.show()
+
+#################################################################
+
+# .. figure:: /_static/img/quickstart/fashion_mnist.png
+#    :alt: fashion_mnist
+#
+
+#################################################################
+# Creating a Custom Dataset
+# -----------------
+#
+# To work with your own data lets look at the a simple custom image Dataset implementation:
```

Comment:
I like your flow of showing the broader picture first and then diving into each component. Can we start with a skeleton of `CustomImageDataset` and then proceed to fill them in + explanations (as you have done below)?

Reply:
I thought about this but I'm on the fence about whether the flow will be cleaner. Will think it over to see the best way to do this.